frontend/src/components/ModuleCard.tsx

```diff
@@ -76,9 +75,13 @@ const ModuleItem = ({ details }: { details: Module }) => {
 
   return (
     <div className="flex h-46 w-full flex-col gap-3 rounded-lg border-1 border-gray-200 p-4 shadow-xs ease-in-out hover:shadow-md dark:border-gray-700 dark:bg-gray-800">
-      <ActionButton onClick={handleClick}>
+      <button
+        type="button"
+        onClick={handleClick}
+        className="text-start font-semibold text-blue-400 hover:underline"
+      >
```

Comment:
Why is this a button? 😮 Somehow missed that before..
This should be just a `Link` element. Works the same and the navigation is actually faster. 
```
<Link
    href={`${window.location.pathname}/modules/${details.key}`}
    className="text-start font-semibold text-blue-400 hover:underline"
  >
    <TruncatedText text={details?.name} />
</Link>
```